src/DurableTask.Netherite/OrchestrationService/OrchestrationMessageBatch.cs

```diff
@@ -118,16 +118,22 @@ public override void OnReadComplete(TrackedObject s, Partition partition)
                 this.workItem.EventCount = historyState.History?.Count ?? 0;
             }
 
-            if (!this.IsExecutableInstance(this.workItem, out var reason))
+            if (!this.IsExecutableInstance(this.workItem, out var reason, out bool traceWarningEvenForTimers))
```

Comment:
nitpick: would it be better to just call this `traceTimerWarnings`?

Reply:
I can see that this is confusing, because of the way the logic is stuctured. I will revise the logic so the decisions are all made inside `this.IsExecutableInstance`.